src/App.tsx

```diff
@@ -8,11 +9,45 @@ const tabs: Tab[] = [
   { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected tab is&nbsp;Tab 1</h1>
-    <div>{tabs.length}</div>
-  </div>
-);
+interface State {
+  tabsList: Tab[],
+  selectedTab: Tab | undefined,
+}
+
+class App extends React.Component<{}, State> {
+  state = {
+    tabsList: [...tabs],
+    selectedTab: tabs[0],
+  };
+
+  onTabSelected = (tab: Tab) => {
+    const { tabsList, selectedTab } = this.state;
+
+    if (tab.id !== selectedTab.id) {
+      this.setState({
+        selectedTab: tabsList.find(desiredTab => desiredTab.id === tab.id),
+      });
+    }
+  };
+
+  render() {
+    const { tabsList, selectedTab } = this.state;
+
+    return (
+      <>
+        <h1>
+          {`${selectedTab.title} is selected`}
+        </h1>
+
+        <Tabs
+          tabs={tabsList}
+          selectedTab={selectedTab}
+          selectedTabId={selectedTab.id}
```

Comment:
Pass only selectedTabId. You add same information to props twice - in tabs and in selectedTab
```suggestion
          selectedTabId={selectedTab.id}
```